GUI/Dialogs/EditLabelsDialog.cs

```diff
@@ -70,14 +79,18 @@ private void LabelSelectionTree_BeforeSelect(Object sender, TreeViewCancelEventA
             {
                 e.Cancel = true;
             }
-            else if (!TryCloseEdit())
+            // This is a proxy for being in LoadTree (it calls ExpandAll)
+            else if (e.Node.IsExpanded)
```

Comment:
Apparently doesn't work on Windows based on testing by @d-faure

Reply:
Replaced with a strategy of removing the `BeforeSelect` event handler and re-adding it.